lib/clean-source-with.nix

```diff
@@ -62,10 +63,32 @@
             && (filter path type && parentFilter path type));
       name' = if name != null
         then name
-        else (if isFiltered && src ? name
-          then src.name
-          else baseNameOf src)
-          + (lib.optionalString (origSubDir != "") ("--" + baseNameOf origSubDir));
+        else
+          if subDirName != ""
+            then if src ? name
+              then src.name + "-" + subDirName 
+              else "cleaned-" + subDirName
```

Comment:
This is a bit circuitous: we have a default value for `name` (null), which we then test for in order to act as if it had a different value ("unnamed"), except sometimes when we use something else ("cleaned"). How about just changing the default value for `name`? i.e.
```
, name ? warnHere "unnamed"
```

Reply:
I felt if you pass in `subDir`, but not `name` it should not warn you since it can derive what is likely to be a reasonable name from the `subDir`.

Reply:
Eh, I'd be happy to just make people name it. But I can see the argument both ways.

Reply:
I think it might soften the impact for people who liked the `baseNameOf src` default and used it carefully.  They can go from:

```
cleanSourceWith { src = ./myDir; }
```

to:

```
cleanSourceWith { src = ./.; subDir = "myDir"; }
```

I prefer this to:

```
cleanSourceWith { src = ./myDir; name = "myDir"; }
```

I know at some point I would have a typo in the `name` or update the `src` and forget to update `name` as well.

Reply:
Okay, I take your argument. But I would go for using "unnamed" as the prefix always just for consistency.

Reply:
Good point.  I've changed it so the prefix is `"unnamed-${caller}-"` (where caller will be "cleanSourceWith" or "cleanGit").  This matches the name given when we issue the warning.  Now the only difference between:

```
cleanSourceWith { src = ./.; subDir = "myDir"; }
```

and

```
cleanSourceWith { src = cleanSourceWith { src = ./.; }; subDir = "myDir"; }
```

should be that the latter will issue a warning to the user about the lack of a name.